boards/stm32l476rg/src/io/i2c.rs

```diff
@@ -45,7 +53,7 @@ impl<'d> HypedI2c for Stm32l476rgI2c<'d> {
 
 impl<'d> Stm32l476rgI2c<'d> {
     /// Create a new instance of our I2C implementation for the STM32L476RG
-    pub fn new(i2c: I2c<'d, Blocking>) -> Self {
+    pub fn new(i2c: Mutex<CriticalSectionRawMutex, RefCell<I2c<'d, Blocking>>>) -> Self {
```

Comment:
can you also just take `I2c<'d, Blocking>` here, and mutex refcell it here?
Then the whole sharing system is completely transparent to the interface user

but i can see that enforcing this in the interface makes it clearer to future i2c implementers that this kind of thing needs to happen

Reply:
actually this isn't a trait method so i'd do the mutexing inside the constructor if that's possible

Reply:
yeah, and pass around the `hyped_i2c` instead? or is that not how it works currently

Reply:
but then `hyped_i2c` would have to be wrapped in the mutex if it's being passed into multiple things?